examples/server.c

```diff
@@ -108,7 +108,9 @@ int main(int argc, char** argv) {
     signal(SIGINT, stopHandler); /* catches ctrl-c */
 
     UA_ServerNetworkLayer nl = UA_ServerNetworkLayerTCP(UA_ConnectionConfig_standard, 16664);
-    UA_ServerConfig config = UA_ServerConfig_standard;
+    UA_ServerConfig config;
+    UA_ServerConfig_standard_new(&config);
```

Comment:
I agree that changing all the config.member to config->member is too much noise for this PR.

The main point is that we should change the public API only once.
We need to break it here to get the "dynamic stuff" into the config.
So let's break it in a way that works for us in the longterm future.

Imho, we can start with your suggestion to
- malloc the config in config_new
- just copy the malloced struct over the server config initially
The latter is purely internal. And is non-critical to adapt at a later time.